askama_derive/src/generator.rs

```diff
@@ -1025,65 +1120,81 @@ impl<'a> Generator<'a> {
                     buf_lit.write(s);
                 };
             }
-            buf.writeln(&format!("writer.write_str({:#?})?;", &buf_lit.buf))?;
-            return Ok(buf_lit.buf.len());
+            let size = buf_lit.buf.len();
+            return Ok((size, Some((buf_lit, None))));
         }
 
+        let mut expr_cache = HashMap::with_capacity(self.buf_writable.len());
+
         let mut size_hint = 0;
         let mut buf_format = Buffer::new(0);
-        let mut buf_expr = Buffer::new(buf.indent + 1);
-        let mut expr_cache = HashMap::with_capacity(self.buf_writable.len());
+        let mut buf_expr = Buffer::new(indent + 1);
+
         for s in mem::take(&mut self.buf_writable) {
             match s {
                 Writable::Lit(s) => {
                     buf_format.write(&s.replace('{', "{{").replace('}', "}}"));
                     size_hint += s.len();
                 }
                 Writable::Expr(s) => {
-                    use self::DisplayWrap::*;
```

Comment:
Why?

Reply:
Ah right, I needed to add the import to the other branch too and then my OCD triggered: it's horrible to have pattern matching with just idents and no path as then my brain doesn't know whether it means "bind everything" or "yep, look at the next one, it's actually binding each variant". I can revert that if you prefer.

Reply:
I can live with it in this case since it's just two arms.